pkgs/applications/emulators/retroarch/point-assets-core-info-paths-to-out.patch

```diff
@@ -0,0 +1,126 @@
+From c2b05fcbde671ea3e30eb285bf6af16699f8b805 Mon Sep 17 00:00:00 2001
+From: Thiago Kenji Okada <thiagokokada@gmail.com>
+Date: Sat, 29 Oct 2022 12:27:55 +0100
+Subject: [PATCH] Point assets/core info paths to $out
+
+---
+ configuration.c                  |  2 +-
+ frontend/drivers/platform_unix.c | 52 ++++++++++++++++----------------
+ 2 files changed, 27 insertions(+), 27 deletions(-)
+
+diff --git a/configuration.c b/configuration.c
+index d9ab8370c4..06a99236c2 100644
+--- a/configuration.c
++++ b/configuration.c
+@@ -1483,7 +1483,7 @@ static struct config_path_setting *populate_settings_path(
+    SETTING_PATH("core_options_path",
+          settings->paths.path_core_options, false, NULL, true);
+    SETTING_PATH("libretro_info_path",
+-         settings->paths.path_libretro_info, false, NULL, true);
++         settings->paths.path_libretro_info, false, NULL, false);
+    SETTING_PATH("content_database_path",
+          settings->paths.path_content_database, false, NULL, true);
+    SETTING_PATH("cheat_database_path",
+diff --git a/frontend/drivers/platform_unix.c b/frontend/drivers/platform_unix.c
+index 4a3b928cb4..c775eb1a76 100644
+--- a/frontend/drivers/platform_unix.c
++++ b/frontend/drivers/platform_unix.c
+@@ -1772,27 +1772,27 @@ static void frontend_unix_get_env(int *argc,
+    fill_pathname_join(g_defaults.dirs[DEFAULT_DIR_CORE_INFO], base_path,
+          "core_info", sizeof(g_defaults.dirs[DEFAULT_DIR_CORE_INFO]));
+ #else
+-   fill_pathname_join(g_defaults.dirs[DEFAULT_DIR_CORE_INFO], base_path,
+-         "cores", sizeof(g_defaults.dirs[DEFAULT_DIR_CORE_INFO]));
++   fill_pathname_join(g_defaults.dirs[DEFAULT_DIR_CORE_INFO], "$out",
++         "share/libretro/info", sizeof(g_defaults.dirs[DEFAULT_DIR_CORE_INFO]));
+ #endif
+    fill_pathname_join(g_defaults.dirs[DEFAULT_DIR_AUTOCONFIG], base_path,
+          "autoconfig", sizeof(g_defaults.dirs[DEFAULT_DIR_AUTOCONFIG]));
+ 
+-   if (path_is_directory("/usr/local/share/retroarch/assets"))
++   if (path_is_directory("@out@/local/share/retroarch/assets"))
```

Comment:
Upstream hardcodes `/usr/local/` as prefix? IMO, making the prefix configurable upstream would be preferred over maintaining this patch in nixpkgs.

Reply:
I can try to upstream this later. For now I will keep the patch.